SingularityUI/app/components/machines/SlaveUsage.jsx

```diff
@@ -0,0 +1,211 @@
+    var grid = props.slaveUsage.map((slave, index) => {
+      return drawSlave(slave, index);
+    });
+    return grid;
+  };
+
+  const drawSlave = (slave, index) => {
+    var criticalGlyph = 'glyphicon glyphicon-remove-sign';
+    var warningGlyph = 'glyphicon glyphicon-minus-sign';
+    var okGlyph = 'glyphicon glyphicon-ok-sign';
+    var criticalStyle = 'danger';
+    var warningStyle = 'warning';
+    var okStyle = 'success';
+
+    if (isStatCritical(slave, props.cpusUsedStat) || isStatCritical(slave, props.memoryBytesUsedStat) || isStatCritical(slave, props.numTasksStat)) {
+      return slaveWithStats(slave, index, criticalStyle, criticalGlyph);
+    } else if (isStatWarning(slave, props.cpusUsedStat) || isStatWarning(slave, props.memoryBytesUsedStat) || isStatWarning(slave, props.numTasksStat)) {
+      return slaveWithStats(slave, index, warningStyle, warningGlyph);
+    } else {
+      return slaveWithStats(slave, index, okStyle, okGlyph);
+    }
+  };
+
+  const isStatWarning = (slave, statName) => {
+    switch (statName) {
+      case props.cpusUsedStat:
+        return (slave.cpusUsed / getMaxAvailableResource(slave, statName)) > props.cpusWarningThreshold;
+      case props.memoryBytesUsedStat:
+        // todo: create util method to convert from mb to bytes
+        return (slave.memoryBytesUsed / (getMaxAvailableResource(slave, statName) * Math.pow(1024, 2))) > props.memoryWarningThreshold;
+      case props.numTasksStat:
+        return slave.numTasks > props.numTasksWarning;
+    }
+  };
+
+  const isStatCritical = (slave, statName) => {
+    switch (statName) {
+      case props.cpusUsedStat:
+        return (slave.cpusUsed / getMaxAvailableResource(slave, statName)) > props.cpusCriticalThreshold;
+      case props.memoryBytesUsedStat:
+        // todo: create util method to convert from mb to bytes
+        return (slave.memoryBytesUsed / (getMaxAvailableResource(slave, statName) * Math.pow(1024, 2))) > props.memoryCriticalThreshold;
+      case props.numTasksStat:
+        return slave.numTasks > props.numTasksWarning;
+    }
+  };
+
+  const getMaxAvailableResource = (slave, statName) => {
+    var slaveInfo = _.findWhere(props.slaves, {'id' : slave.slaveId});
+    
+    switch (statName) {
+      case props.cpusUsedStat:
+        try {
+          return parseFloat(slaveInfo.attributes.real_cpus) || slaveInfo.resources.cpus;
+        } catch (e) {
+          throw 'Could not find resources (cpus) for slave ' + slave.slaveId;
+        }
+      case props.memoryBytesUsedStat:
+        try {
+          return parseFloat(slaveInfo.attributes.real_memory_mb) || slaveInfo.resources.mem;
+        } catch (e) {
+          throw 'Could not find resources (memory) for slave ' + slave.slaveId;
+        }
+      default:
+        throw statName + ' is an unsupported statistic';
+    }
+  };
+
+  const slaveWithStats = (slave, index, bsStyle, glyphicon) => (
+    <Dropdown key={slave.slaveId} id={index.toString()}>
```

Comment:
The dropdown is the little stat menu you get when clicking on a slave btn. Using Dropdown over DropdownMenu allows more styling flexibility